packages/react-dom/test/index.test.tsx

```diff
@@ -13,38 +13,37 @@ test('`x` and `y` are initially `null`', async () => {
   expect(result.current.y).toBe(null);
 });
 
-test('`middleware` is memoized internally', async () => {
-  function Component() {
-    const {reference, floating} = useFloating({
-      middleware: [
```

Comment:
@KubaJastrz I think we don't even need to test for the infinite loop because if there is one then the test will break anyway.